### ISSUE TITLE
Failed to write session data (redis cluster mode)

### ISSUE THREAD

**Issue by zibuyule** [phpredis/phpredis#1782]
### Expected behaviour
Successfully logged into phpmyadmin

### Actual behaviour
Can't log in to phpmyadmin
![access phpmyadmin](https://user-images.githubusercontent.com/5792519/83968378-68fead00-a8fb-11ea-8d83-765e711f4501.jpg)

### I'm seeing this behaviour on
- OS:  ubuntu18.04 
- Redis: 5.0.6 (enable cluster mode)
- PHP: 7.2.25
- phpredis: 5.1.1
- phpmyadmin: 4.9.5

### Steps to reproduce, backtrace or example script
![php ini-session](https://user-images.githubusercontent.com/5792519/83968254-736c7700-a8fa-11ea-9470-aaba6f3b37d0.jpg)

### I've checked
- [ ] There is no similar issue from other users
- [x] Issue isn't fixed in `develop` branch

**Comment by michael-grunder** [comment]
We have signed up for an AWS account so we can hopefully be more helpful with these kinds of issues.

It will take some time as neither @yatsukhnenko or myself have ever really used AWS but hopefully soon!

**Comment by zibuyule** [comment]
Hi Michael,

Could you please provide an update?

Thanks!

**Comment by yatsukhnenko** [comment]
@zibuyule Could you test the following patch?
```patch
diff --git a/redis_session.c b/redis_session.c
index 0e72fd4..4165fff 100644
--- a/redis_session.c
+++ b/redis_session.c
@@ -854,7 +854,7 @@ static char *cluster_session_key(redisCluster *c, const char *key, int keylen,
 
 PS_OPEN_FUNC(rediscluster) {
     redisCluster *c;
-    zval z_conf, *zv;
+    zval z_conf, *zv, *context;
     HashTable *ht_conf, *ht_seeds;
     double timeout = 0, read_timeout = 0;
     int persistent = 0, failover = REDIS_FAILOVER_NONE;
@@ -932,6 +932,10 @@ PS_OPEN_FUNC(rediscluster) {
 
     redis_sock_set_auth(c->flags, user, pass);
 
+    if ((context = REDIS_HASH_STR_FIND_TYPE_STATIC(ht_conf, "stream", IS_ARRAY)) != NULL) {
+        redis_sock_set_stream_context(c->flags, context);
+    }
+
     /* First attempt to load from cache */
     if (CLUSTER_CACHING_ENABLED()) {
         hash = cluster_hash_seeds(seeds, nseeds);
```

**Comment by yatsukhnenko** [comment]
```ini
session.save_handler = rediscluster
session.save_path = "seed[]=tls://ip1:port&seed[]=tls://ip2:port&stream[verify_peer]=0&stream[local_cert]=file:///path/to/cert.pem"
```

**Comment by nlalexandrov** [comment]
Tested, looks good, works as expected.
Thank you for your work!

**Comment by yatsukhnenko** [comment]
Fixed in 5.3.2

**Comment by nusphere** [comment]
this 

> ```ini
> session.save_handler = rediscluster
> session.save_path = "seed[]=tls://ip1:port&seed[]=tls://ip2:port&stream[verify_peer]=0&stream[local_cert]=file:///path/to/cert.pem"
> ```

this is only available for cluster?

```
+    if ((context = REDIS_HASH_STR_FIND_TYPE_STATIC(ht_conf, "stream", IS_ARRAY)) != NULL) {
+        redis_sock_set_stream_context(c->flags, context);
+    }
```

is only implemented at rediscluster
